api/node/__test__/api.spec.ts

```diff
@@ -55,3 +61,49 @@ test('constructor parameters', (t) => {
     t.is(test.check, "test");
     t.is(hello, "hello");
 })
+
+
+test('callback closure cyclic references do not prevent GC', async (t) => {
+
+    // Setup:
+    // A component instance with a callback installed from JS:
+    //    * The callback captures the surrounding environment, which
+    //      includes an extra reference to the component instance itself
+    //      --> a cyclic reference
+    //    * Invoking the callback clears the reference in the outer captured
+    //      environment.
+    //
+    // Note: WeakRef's deref is used to observe the GC. This means that we must
+    // separate the test into different jobs with await, to permit for collection.
+    // (See https://tc39.es/ecma262/multipage/managing-memory.html#sec-weak-ref.prototype.deref)
+
+    let demo = loadFile(path.join(__dirname, "resources/test-constructor.slint")) as any;
+    let callback_invoked = false;
```

Comment:
Stupid bug here: The instance is never created! The `my_callback` assignments happen to work because well, it's just a JS object. And because we don't have type safety, the compiler didn't warn us. So the test never really tested what it should have: instantiate the Rust code that presumably keeps a strong ref.